pyvista/core/dataset.py

```diff
@@ -2115,6 +2115,24 @@ def overwrite(self, mesh: _vtk.vtkDataSet):
         if is_pyvista_dataset(mesh):
             self.copy_meta_from(mesh, deep=True)
 
+    def overwrite(self, mesh: _vtk.vtkDataSet):
+        """Overwrite this dataset inplace with the new dataset's geometries and data.
+
+        .. deprecated:: 0.37.0
+            Use :attr:`DataSet.copy_from` instead.
+
+        Parameters
+        ----------
+        mesh : vtk.vtkDataSet
+            The overwriting mesh.
+
+        """
+        warnings.warn(
+            "Use of `DataSet.overwrite` is deprecated. Use `DataSet.copy_from` instead.",
+            PyvistaDeprecationWarning,
+        )
+        return self.copy_from(mesh)
```

Comment:
numpydoc will require a `Returns` section when you return something.  `copy_from` doesn't return, so we can do that here too.  Or, a section could be added in the docstring.

```suggestion
        self.copy_from(mesh)
```